presto-hive/src/main/java/com/facebook/presto/hive/HivePartialAggregationPushdown.java

```diff
@@ -182,7 +182,7 @@ private boolean isAggregationPushdownSupported(AggregationNode partialAggregatio
                         return false;
                     }
 
-                    if (hiveStorageFormat == ORC) {
+                    if (hiveStorageFormat == ORC || hiveStorageFormat == DWRF) {
```

Comment:
Since the supported datatypes are specific to the file format, I think it might be better to split this out by file format so there is less chance for error when adding support for a new file format here (there is overlap between orc and parquet support, but it seems more incidental to me that they happen to have some of the same restrictions). e.g. something like
```
switch(hiveStorageFormat){
case PARQUET:
   return isSupportedParquetTypeForMinMax(type);
case ORC:
case DWRF:
    return isSupportedOrcTypeForMinMax(type);
default:
    throw exception
}
```

Reply:
> also, can't comment there, but have you tested that isPartialAggregationPushdownForVariableLengthDatatypesEnabled works for DWRF format?

tested it now.
with flag turned on https://www.internalfb.com/intern/presto/query/?query_id=20230823_213412_60457_pyr5r#plan completely uses footer information .
without flag https://www.internalfb.com/intern/presto/query/?query_id=20230823_213319_60417_pyr5r#plan has to scan all rows.

the unit test will cover this as well.

Reply:
Added Integration test and refactored the code.